tests/queries/0_stateless/02158_groupsortedarray.reference

```diff
@@ -0,0 +1,12 @@
+[0,1,2,3,4]
+[0,1,2,3,4,5,6,7,8,9]
+[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99]
+[999,998,997,996,995,994,993,992,991,990,989,988,987,986,985,984,983,982,981,980,979,978,977,976,975,974,973,972,971,970,969,968,967,966,965,964,963,962,961,960,959,958,957,956,955,954,953,952,951,950,949,948,947,946,945,944,943,942,941,940,939,938,937,936,935,934,933,932,931,930,929,928,927,926,925,924,923,922,921,920,919,918,917,916,915,914,913,912,911,910,909,908,907,906,905,904,903,902,901,0]
```

Comment:
This output is clearly wrong, no way 0 follows 901:

```
SELECT groupSortedArrayWeighted(5)(number, -number)
FROM numbers(1000)

Query id: f2c986a7-c5c4-4296-a39e-928dd5ba3f26

┌─groupSortedArrayWeighted(5)(number, negate(number))─┐
│ [999,998,997,996,0]                                 │
└─────────────────────────────────────────────────────┘
```

Reply:
This is happening because we always used UInt64 for weight comparison, I have added another template argument to use signed/unsiged depending on the column type.